x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_usage.yml

```diff
@@ -92,7 +92,7 @@ setup:
   - gt: {esql.functions.to_long: $functions_to_long}
   - match: {esql.functions.coalesce: $functions_coalesce}
   # Testing for the entire function set isn't feasbile, so we just check that we return the correct count as an approximation.
-  - length: {esql.functions: 127} # check the "sister" test below for a likely update to the same esql.functions length check
+  - length: {esql.functions: 128} # check the "sister" test below for a likely update to the same esql.functions length check
```

Comment:
I thought we'd blasted this thing....

Reply:
We didn't (we should IMHO) and it keeps failing.
@idegtiarenko please also update the counter at line 166, otherwise it will fail in release tests, and you won't realize it until it's merged.

Reply:
I am a bit surprised about the other one. 
Currently it is passing in the pr, meaning the count is expected (for snapshot build).
Is there a way to match against different count in release and non release builds? 
Or may be this is an argument against `(Build.current().isSnapshot())`? (see https://github.com/elastic/elasticsearch/pull/117989#discussion_r1869988231)

Reply:
The PR is green because the second one is just not running. You can make it run adding `test-release` label to the PR, but the CI will take much more to run.

`(Build.current().isSnapshot())` in capabilities only impacts on tests, if you want to make the function as snapshot-only you'll have to register it as a [snapshot function in EsqlFunctionRegistry](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistry.java#L420-L428) rather than a normal one.

Reply:
@idegtiarenko those two tests are executed depending on snapshot nature of the build. [One test runs in snapshot-only](https://github.com/elastic/elasticsearch/blob/48d4dd120c949ebad62d58ed01ab44bf804499d8/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_usage.yml#L33) CI runs, the [other one runs in release-only](https://github.com/elastic/elasticsearch/blob/48d4dd120c949ebad62d58ed01ab44bf804499d8/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_usage.yml#L105) CI runs (using `test-release` label on the PR). The different nature of those two tests is visible in the capabilities settings.

Reply:
Thanks all!